app/assets/javascripts/discourse/app/components/d-modal.js

```diff
@@ -206,9 +206,8 @@ export default class DModal extends Component {
         // attempt to focus the first of the focusable elements or just the modal-body
         // to make it possible to scroll with arrow down/up
         (
-          innerContainer.querySelector(
-            focusableElements + ", button:not(.modal-close)"
```

Comment:
We set `tabindex=-1` on the `.modal-close` button instead of adding yet another condition here in the query selector. This was actually causing a subtle bug because any button even with the `tabindex=-1` attribute will still be matched.